### PR TITLE
AL/BE-86. Accounts pallet.

### PR DESCRIPTION
**Pull request by lut1k**
Added accounts pallet to runtime.
Added add_member and remove_member functions to accounts pallet.
Implemented unit-tests for add_member and remove_member  functions.

### CODE REVIEW

File: pallets/accounts/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		}
+
+		/// Remove an account from the allow-list.
+		///
+		/// The dispatch origin of this call must be _Root_.
+		#[weight = 0]
+		fn remove_member(origin, account_to_remove: T::AccountId) -> DispatchResult {
+			ensure_root(origin)?;
+
+			ensure!(AllowedAccounts::<T>::contains_key(&account_to_remove), Error::<T>::NotMember);
```
Comment (Pavel-Blaize) [inline pallets/accounts/src/lib.rs:78]: Add one more check - if single admin has left, than it cannot be removed.
We need it to ensure, that the node will not be frozen without any admins, as these methods in the future will be allowed for admins only.

File: pallets/accounts/src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		// Try to add the 17th member exceeding the max
+		assert_noop!(
+			TestAccounts::add_member(Origin::root(), 16),
+			Error::<Test>::MembershipLimitReached
+		);
+	})
+}
+
+#[test]
+fn remove_member_should_work() {
```
Comment (Pavel-Blaize) [inline pallets/accounts/src/tests.rs:43]: Also, based on the upper comment, add test, that the last admin cannot be removed.

File: pallets/accounts/src/lib.rs (hunk 1 above)
Comment (lut1k) [reply pallets/accounts/src/lib.rs:78]: > Add one more check - if single admin has left, than it cannot be removed.
> We need it to ensure, that the node will not be frozen without any admins, as these methods in the future will be allowed for admins only.

fixed

File: pallets/accounts/src/tests.rs (hunk 2 above)
Comment (lut1k) [reply pallets/accounts/src/tests.rs:43]: > Also, based on the upper comment, add test, that the last admin cannot be removed.

Fixed.

**Review by Pavel-Blaize — APPROVED** [review]

**Review by Apollo50 — APPROVED** [review]

**Pull request merged**